filters/f_swresample.c

```diff
@@ -450,15 +454,22 @@ static struct mp_frame filter_resample_output(struct priv *p,
     if (!p->avrctx)
         goto error;
 
-    int samples = get_out_samples(p, in ? mp_aframe_get_size(in) : 0);
+    // Limit the filtered data size for better latency when changing speed.
+    // Avoid buffering data within the resampler => restrict input size.
+    double s = p->opts->max_output_frame_size / 1000;
+    int max_in = MPCLAMP(s * p->in_rate/ p->out_rate * p->in_rate, 128, INT_MAX);
```

Comment:
Since max_output_frame_size is time-based, doesn't it just need to be multiplied by the input rate to get the number of input frames for that duration? Or am I misunderstanding what this is calculating?

Reply:
It includes the speed change, so if you e.g. play twice as fast, the input needs to be double the size to output a fragment of a certain actual duration. It tries to do that by using the sample rate ratio.